servicetalk-opentelemetry-http/src/main/java/io/servicetalk/opentelemetry/http/AbstractOpenTelemetryHttpRequesterFilter.java

```diff
@@ -53,7 +58,7 @@ public StreamingHttpClientFilter create(FilterableStreamingHttpClient client) {
             @Override
             protected Single<StreamingHttpResponse> request(final StreamingHttpRequester delegate,
                                                             final StreamingHttpRequest request) {
-                return Single.defer(() -> trackRequest(delegate, request, null).shareContextOnSubscribe());
+                return Single.defer(() -> trackRequest(delegate, request, null, false).shareContextOnSubscribe());
```

Comment:
Wanna highlight a hidden feature: there is also `StreamingHttpClientFilter.reserveConnection(...)` method that by default executes what's implemented in `request` for a reserved connection. Would you like to keep things as-is or implement `reserveConnection` to flip the `connectionLevel` flag to `true`? Either way it will work, it's just a question of which `if` branch we prefer to execute: the one that calls `current.get(SHOULD_INSTRUMENT_KEY)` and always falls back to `shouldStart` or the one that is wastefully sets a new `Scope` with a key that will never be used? I would tend to fewer memory allocations on the requests path if possible.

Reply:
I think in terms of correctness, we need to leave it alone, don't we? If someone uses the filter in both logical and physical request position, I think we want the result to be the same as if they'd made the simple request because they are traversing the whole filter pipeline, only in a way that uses the same connection.